integrations/mongodb_atlas/src/haystack_integrations/document_stores/mongodb_atlas/document_store.py

```diff
@@ -488,16 +458,8 @@ def _embedding_retrieval(
                     "filter": filters,
                 }
             },
-            {
-                "$project": {
-                    "_id": 0,
-                    "content": 1,
-                    "blob": 1,
-                    "meta": 1,
-                    self.embedding_field: 1,
-                    "score": {"$meta": "vectorSearchScore"},
-                }
-            },
+            {"$addFields": {"score": {"$meta": "vectorSearchScore"}}},
+            {"$project": {"_id": 0}},
```

Comment:
Can you please explain this change? I would like to understand it correctly...

Reply:
Instead of selecting some fields that are returned i am returning the whole object in Mongodb but: 

- exclude the internal _id (as before)
- add the score manually from vector search

this ensures that metadata loading works